src/event/event.rs

```diff
@@ -58,30 +69,72 @@ impl Event {
     /// Method is available on all platforms, but not all platforms (can) use
     /// this indicator.
     ///
-    /// When HUP events are received is different per platform. For example on
-    /// epoll platforms HUP will be received when a TCP socket reached EOF,
-    /// however on kqueue platforms a HUP event will be received when `shutdown`
-    /// is called on the connection (both when shutting down the reading
-    /// **and/or** writing side). Meaning that even though this function might
-    /// return true on one platform in a given situation, it might return false
-    /// on a different platform in the same situation. Furthermore even if true
-    /// was returned on both platforms it could simply mean something different
-    /// on the two platforms.
-    ///
     /// Because of the above be cautions when using this in cross-platform
     /// applications, Mio makes no attempt at normalising this indicator and
-    /// only provides a convenience method to read it.
+    /// only provides a convenience method to read it. We advice looking at the
+    /// documentation provided for the selectors (see below) when using this
+    /// indicator.
+    ///
+    /// The table below shows what flags are checked on what OS.
+    ///
+    /// | [OS selector] | Flag(s) checked |
+    /// |---------------|-----------------|
+    /// | [epoll]       | `EPOLLHUP`      |
+    /// | [kqueue]      | Not supported   |
+    ///
+    /// [OS selector]: ../struct.Poll.html#implementation-notes
+    /// [epoll]: http://man7.org/linux/man-pages/man7/epoll.7.html
+    /// [kqueue]: https://www.freebsd.org/cgi/man.cgi?query=kqueue&sektion=2
     #[inline]
     pub fn is_hup(&self) -> bool {
         sys::event::is_hup(&self.inner)
     }
 
+    /// Returns true if the event contains read HUP readiness.
+    ///
+    /// # Notes
+    ///
+    /// Method is available on all platforms, but not all platforms (can) use
+    /// this indicator.
+    ///
+    /// Because of the above be cautions when using this in cross-platform
+    /// applications, Mio makes no attempt at normalising this indicator and
+    /// only provides a convenience method to read it. We advice looking at the
+    /// documentation provided for the selectors (see below) when using this
+    /// indicator.
+    ///
+    /// The table below shows what flags are checked on what OS.
+    ///
+    /// | [OS selector] | Flag(s) checked |
+    /// |---------------|-----------------|
+    /// | [epoll]       | `EPOLLRDHUP`    |
+    /// | [kqueue]      | `EV_EOF`        |
```

Comment:
I think it is: `EVFILT_READ` with `EV_EOF`.

Reply:
You're right but doesn't the read part of the method name imply that? I'm fine with adding it though.